### PR TITLE
R4R: Add memo check by regexp

### PR DESCRIPTION
**Pull request by chengwenxi**
Resolves: https://github.com/irisnet/irishub/issues/1867

### CODE REVIEW

File: app/v1/bank/keeper.go [hunk 1]
```diff
@@ -110,6 +110,23 @@ func (keeper BaseKeeper) DecreaseLoosenToken(
 	keeper.am.DecreaseTotalLoosenToken(ctx, amt)
 }
 
+// SetMemoRegexp set memo regexp for sender account
+func (keeper BaseKeeper) SetMemoRegexp(ctx sdk.Context, fromAddr sdk.AccAddress, regexp string) (sdk.Tags, sdk.Error) {
+	acc := keeper.am.GetAccount(ctx, fromAddr)
+	if acc == nil {
+		acc = keeper.am.NewAccountWithAddress(ctx, fromAddr)
+	}
```
Comment (chengwenxi) [inline app/v1/bank/keeper.go]: Return error, not create a new address.

File: app/v1/bank/keeper.go [hunk 2]
```diff
@@ -110,6 +110,23 @@ func (keeper BaseKeeper) DecreaseLoosenToken(
...
+
+	acc.SetMemoRegexp(regexp)
+	keeper.am.SetAccount(ctx, acc)
+
+	tags := sdk.NewTags(
+		"memoRegexp", []byte(regexp),
```
Comment (chengwenxi) [inline app/v1/bank/keeper.go]: memo-regexp

File: app/v1/auth/feeauth.go [hunk 3]
```diff
@@ -76,6 +76,10 @@ func NewFeeRefundHandler(am AccountKeeper, fk FeeKeeper) types.FeeRefundHandler
 			return sdk.Coin{}, err
 		}
 
+		// set mem regexp
```
Comment (zhiqiang-bianjie) [inline app/v1/auth/feeauth.go]: logical redundancy

File: app/v1/bank/msgs.go [hunk 4]
```diff
@@ -217,3 +220,52 @@ func (msg MsgBurn) GetSignBytes() []byte {
...
+
+// Implements Msg.
+func (msg MsgSetMemoRegexp) ValidateBasic() sdk.Error {
+	if len(msg.Owner) == 0 {
+		return sdk.ErrInvalidAddress(msg.Owner.String())
+	}
+	if _, err := regexp.Compile(msg.MemoRegexp); err != nil {
+		return ErrInvalidMemoRegexp(DefaultCodespace, "invalid memo regexp")
+	}
+	if len(msg.MemoRegexp) > memoRegexpLengthLimit {
```
Comment (zhiqiang-bianjie) [inline app/v1/bank/msgs.go]: 250 lines and 253 lines change order

**Pull request merged**